File: api/db/models/disponibilidade.py
from peewee import CharField, TimeField, ForeignKeyField, BooleanField
from db.models.paciente import Paciente
from db.db import BaseModel

class Disponibilidade(BaseModel):
    dia_semana = CharField()  # Por exemplo, "segunda-feira"
    horario_inicio = TimeField()  # Por exemplo, "09:00"
    horario_fim = TimeField()  # Por exemplo, "12:00"
    is_disponivel = BooleanField(default=True)  # "sim" ou "nao"

File: agenda-psicologo-frontend/src/app/class/Disponibilidade.ts
import { Paciente } from "./Paciente";

export class Disponibilidade {
  id?: number;
  paciente: Paciente; // A propriedade 'paciente' é do tipo 'Paciente'
  dia_semana: string;
  horario_inicio: string;
  horario_fim: string;
  is_disponivel: boolean;

  constructor(paciente: Paciente, dia_semana: string, horario_inicio: string, horario_fim: string, is_disponivel: boolean) {
    this.paciente = paciente; // Atribui a instância do Paciente ao atributo 'paciente'
    this.dia_semana = dia_semana;
    this.horario_inicio = horario_inicio;
    this.horario_fim = horario_fim;
    this.is_disponivel = is_disponivel;
  }
}
